### PR TITLE
mu: update to 1.6.3 & combine mu and mu4e packages

### PR DESCRIPTION
**Pull request by emacsomancer**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by sgn** [comment]
In my opinion, there're no points in spliting `mu4e` from `mu`.

**Comment by emacsomancer** [comment]
I don't have any particular opinion one way or other (I "inherited" the package in this split fashion.)

Two thoughts:

1. I'm not sure what the implications of merging the two might be for arm/crossbuilds.
2. I think it would be far better for a merged package to be called `mu4e` rather than `mu` (since the former is (a) user-facing piece; (b) what people will general be searching for; (c) long enough to avoid name-collisions).

**Comment by sgn** [comment]
> I don't have any particular opinion one way or other (I "inherited" the package in this split fashion.)
> 
> Two thoughts:
> 
>     1. I'm not sure what the implications of merging the two might be for arm/crossbuilds.

`mu4e` was disabled for cross build because `emacs` wasn't available there.
However, `mu4e` is simply text files, and anyone that have it should have emacs installed already.

I'm in favor of merging because someone who wants to use mu4e on (let's say) aarch64, can just build emacs themselves and install `mu` (or `mu4e`) and use it, instead of rebuild `mu`.

>     2. I think it would be far better for a merged package to be called `mu4e` rather than `mu` (since the former is (a) the name of the user-facing piece; (b) what people will general be searching for; (c) long enough to avoid name-collisions).

I have no opinion, either way. One thing: someone who only have mu installed (who are they anyway?) won't see the upgrade from `mu` to `mu4e`.

**Comment by emacsomancer** [comment]
> `mu4e` was disabled for cross build because `emacs` wasn't available there.
> However, `mu4e` is simply text files, and anyone that have it should have emacs installed already.

So then could all of the `if [ ! "$CROSS_BUILD" ]` conditionals just be removed?

What about `emacs` as a dependency?

> > 2. I think it would be far better for a merged package to be called `mu4e` rather than `mu`....

> I have no opinion, either way. One thing: someone who only have mu installed (who are they anyway?) won't see the upgrade from `mu` to `mu4e`.

I can't really think of a use for `mu` without `mu4e`, but in theory I suppose someone could just have `mu` installed.

But:

````
$ xbps-query -Rs mu | wc -l
1080

$ xbps-query -Rs mu4e | wc -l
1
````

**Comment by sgn** [comment]
On 2021-08-08 06:42:50-0700, Benjamin Slade ***@***.***> wrote:
> > `mu4e` was disabled for cross build because `emacs` wasn't available there.
> > However, `mu4e` is simply text files, and anyone that have it should have emacs installed already.
> 
> So then could all of the `if [ ! "$CROSS_BUILD" ]` conditionals just be removed?

Yes, I would say.

> What about `emacs` as a dependency?

I would ignore that dependencies,
whoever uses mu4e should have emacs installed themselves, IMHO.

> 
> > > 2. I think it would be far better for a merged package to be called `mu4e` rather than `mu`....
> 
> > I have no opinion, either way. One thing: someone who only have mu installed (who are they anyway?) won't see the upgrade from `mu` to `mu4e`.
> 
> I can't really think of a use for `mu` without `mu4e`, but in theory I suppose someone could just have `mu` installed.
> 
> But:
> 
> ````
> $ xbps-query -Rs mu | wc -l
> 1080
> 
> $ xbps-query -Rs mu4e | wc -l
> 1
> ````

Fine by me.

Or just keep it as is.

**Comment by emacsomancer** [comment]
Out of curiosity, I looked to see what other distros do.

[Debian](https://packages.debian.org/unstable/mu4e) does maintain `mu` and `mu4e` as separate packages (they also package `mu` as `maildir-utils`, presumably for the sort of reasons discussed above). I don't know if they know something I don't, in terms of keeping these separate. 

But Fedora seems to have them combined (under a package name `[maildir-utils](https://src.fedoraproject.org/rpms/maildir-utils)`), as does Arch's AUR (though under the name `[mu](https://aur.archlinux.org/packages/mu/)`).

So it's probably reasonable to combine them, though I think `mu4e` is the ideal package name for discoverability. I'll work on doing so.

**Comment by emacsomancer** [comment]
(Let me know if there's something else I should do at this point.)

**Comment by emacsomancer** [comment]
Changing some things, in part because 1.6.3 was released.

File: srcpkgs/mu4e/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+pre_configure() {
+	sed -i 's,-I${prefix}/include,,' contrib/Makefile.am
+	autoreconf -fi
+}
+
+pkg_install() {
+	vmove usr/share/emacs
+	vmove usr/share/info
+	vmove usr/share/doc/mu/mu4e*
+}
```
Comment (sgn) [inline srcpkgs/mu4e/template:26]: This is no longer needed.

File: srcpkgs/mu4e/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+build_style=gnu-configure
+configure_args="--enable-mu4e $(vopt_if guile --enable-guile)"
+hostmakedepends="automake emacs libtool pkg-config texinfo glib-devel"
+makedepends="xapian-core-devel gmime3-devel libuuid-devel $(vopt_if guile guile)"
+short_desc="Maildir-utils indexer/searcher and associated Emacs client"
+maintainer="Benjamin Slade <slade@lambda-y.net>"
+license="GPL-3.0-or-later"
+homepage="https://www.djcbsoftware.nl/code/mu/"
+distfiles="https://github.com/djcb/mu/releases/download/${version}/mu-${version}.tar.xz"
+checksum=fe286eebc2d9a2060a67c365aed7dd7a6de1fe104f89da36976d8737200754b6
```
Comment (sgn) [inline srcpkgs/mu4e/template:15]:
```sh
replaces="mu<${version}"
provides="mu-${version}_${revision}"
```

File: srcpkgs/mu4e/template (hunk 1 above)
Comment (emacsomancer) [reply srcpkgs/mu4e/template]: removed

File: srcpkgs/mu4e/template (hunk 2 above)
Comment (emacsomancer) [reply srcpkgs/mu4e/template:15]: added

**Pull request merged**